src/journal/compress.c

```diff
@@ -541,6 +645,90 @@ int compress_stream_lz4(int fdf, int fdt, uint64_t max_bytes) {
 #endif
 }
 
+int compress_stream_zstd(int fdf, int fdt, uint64_t max_bytes) {
+#if HAVE_ZSTD
+        _cleanup_(ZSTD_freeCCtxp) ZSTD_CCtx *s = ZSTD_createCStream();
+        size_t out_size = ZSTD_CStreamOutSize();
```

Comment:
Note that `out_size` is just over 128 KB, which is a lot to allocate on the stack. You could use a smaller size.